pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherSingleActiveConsumer.java

```diff
@@ -626,4 +633,57 @@ public void recycle() {
             recyclerHandle.recycle(this);
         }
     }
+
+    public static void readCompactedEntries(TopicCompactionService topicCompactionService, ManagedCursor cursor,
```

Comment:
Why did you make it public? It should be private.

Reply:
Because it needs to be used by Kop.

Reply:
How about moving this logic into the `Dispatchers` util class? Because it's a static method, and it's weird for the eco-system to call it in [PersistentDispatcherSingleActiveConsumer.java](https://github.com/apache/pulsar/pull/20718/files?diff=split&w=1#diff-cf3d6c439d725b9688d46adb393e4f36ec0470b201101c1f38e0e90d145cf0b6) :)

Reply:
Good idea.